README.md

```diff
@@ -4,32 +4,391 @@
+
+```bash
+juju deploy loki-k8s --channel=stable
+```
+
+
+### Checking deployment status
+
+Once the Charmed Operator is deployed, the status can be checked running:
+
+```bash
+juju status --color --relations
+```
+
+
+```bash
+$ juju status --color --relations
+
+Model          Controller           Cloud/Region        Version  SLA          Timestamp
+observability  charm-dev-batteries  microk8s/localhost  3.0.2    unsupported  10:39:52-03:00
+
+App       Version  Status   Scale  Charm     Channel  Rev  Address         Exposed  Message
+loki-k8s  2.4.1    waiting      1  loki-k8s  stable    47  10.152.183.195  no       waiting for container
+
+Unit         Workload  Agent  Address      Ports  Message
+loki-k8s/0*  active    idle   10.1.36.115
+```
+
+
+
+### Loki HTTP API
+
+Loki Charmed Operator exposes its [HTTP API](https://grafana.com/docs/loki/latest/api/) over port 3100.
+
+
+#### Example 1 - Get Loki version:
+
+`/loki/api/v1/status/buildinfo` exposes the build information in a JSON object.
+The fields are `version`, `revision`, `branch`, `buildDate`, `buildUser`, and `goVersion`.
+
+```bash
+loki_ip=$(juju status loki-k8s/0 | grep "loki-k8s/0" | awk '{print $4}')
+
+curl http://$loki_ip:3100/loki/api/v1/status/buildinfo
+{"version":"2.4.1","revision":"f61a4d261","branch":"HEAD","buildUser":"root@39a6e600b2df","buildDate":"2021-11-08T13:09:51Z","goVersion":""}
+```
+
+
+
+#### Example 2 - Send logs entries to Loki with curl:
+
+`/loki/api/v1/push` is the endpoint used to send log entries to Loki. The default behavior is for the POST body to be a snappy-compressed protobuf message.
+Alternatively, if the `Content-Type` header is set to `application/json`, a JSON post body can be sent in the following format:
+
+```bash
+loki_ip=$(juju status loki-k8s/0 | grep "loki-k8s/0" | awk '{print $4}')
+
+curl -v -H "Content-Type: application/json" -XPOST -s "http://$loki_ip:3100/loki/api/v1/push" --data-raw \
+  '{"streams": [{ "stream": { "foo": "bar2" }, "values": [ [ "1570818238000000000", "fizzbuzz" ] ] }]}'
+```
+
+
+#### Example 3 - Send logs entries to Loki with Promtail:
+
+[Promtail](https://grafana.com/docs/loki/latest/clients/promtail/) is an agent which ships the contents of local logs to Loki. It is usually deployed to every machine that has applications needed to be monitored.
+
+It primarily:
+
+- Discovers targets
+- Attaches labels to log streams
+- Pushes them to the Loki instance.
+
+Currently, Promtail can tail logs from two sources: local log files and the systemd journal (on AMD64 machines only).
+
+To set up a Promtail instance to work with Loki Charmed Operator please refer to [Configuring Promtail documentation](https://grafana.com/docs/loki/latest/clients/promtail/configuration/). Anyway the most important part is the `clients` section in Promtail config file, for instance:
+
+```yaml
+clients:
+  - url: http://<LOKI_ADDRESS>:3100/loki/api/v1/push
+```
+
+
+
+## Relations
```

Comment:
Now that we (sometimes) use [`description:` key in metadata.yaml](https://github.com/canonical/traefik-k8s-operator/blob/d8adda8e4eec41e270f403fbaa9ec56a74bd62d0/metadata.yaml#L54), should we incorporate this section in metadata instead?

Reply:
From my point of view we can add some description to metadata.yaml, but these explanations are to long to be there.

What do you think @simskij ?

Reply:
I agree with you, José.